### PR TITLE
[FSSDK-10616] overall test improvement

### PR DESCRIPTION
**Pull request by junaed-optimizely**
## Summary
- Unit tests improvements of all the modules except ["Client & Provider"](https://jira.sso.episerver.net/browse/FSSDK-10439)
- 33 new tests have been added 
- Improvements & deprecated API updates

## Test plan
- 33 new tests have been added along with improvement & deprecated API replacement

## Issues
- [FSSDK-10616](https://jira.sso.episerver.net/browse/FSSDK-10616)

### CODE REVIEW

File: src/utils.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+import { UserInfo } from './utils';
+
+describe('utils', () => {
+  describe('areUsersEqual', () => {
+    const user = { id: '1', attributes: { name: 'user1' } };
+
+    it('returns true if users are equal', () => {
+      const user2 = JSON.parse(JSON.stringify(user));
+
+      expect(utils.areUsersEqual(user, user2)).toBe(true);
```
Comment (mikechu-optimizely) [inline src/utils.spec.tsx:29]: NIT: Is it clearer to separate out the "act"? 

```suggestion
      const user2 = JSON.parse(JSON.stringify(user));
      
      const usersAreEqual = utils.areUsersEqual(user, user2)

      expect(usersAreEqual).toBe(true);
```

**Review by mikechu-optimizely — APPROVED** [review]
LGTM

File: src/utils.spec.tsx (hunk 1 above)
Comment (junaed-optimizely) [reply src/utils.spec.tsx:29]: Yes. we can do that. That would be more readable. Let me update! Thanks for the suggestion

**Review by mikechu-optimizely — APPROVED** [review]
Thanks for the added coverage. LGTM

**Pull request merged**